File: backend/src/main/java/com/lawfirm/domain/admin/entity/Supplier.java
package com.lawfirm.domain.admin.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.lawfirm.common.base.BaseEntity;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

/**
 * 供应商实体
 */
@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@TableName("admin_supplier")
public class Supplier extends BaseEntity {

    /**
     * 供应商编号
     */
    private String supplierNo;

    /**
     * 供应商名称
     */
    private String name;

    /**
     * 供应商类型：GOODS-物品供应商, SERVICE-服务供应商, BOTH-综合供应商
     */
    private String supplierType;

    /**
     * 联系人
     */
    private String contactPerson;

    /**
     * 联系电话
     */
    private String contactPhone;

    /**
     * 联系邮箱
     */
    private String contactEmail;

    /**
     * 地址
     */
    private String address;

    /**
     * 统一社会信用代码
     */
    private String creditCode;

    /**
     * 开户银行
     */
    private String bankName;

    /**
     * 银行账号
     */
    private String bankAccount;

    /**
     * 供应范围
     */
    private String supplyScope;

    /**
     * 评级：A-优秀, B-良好, C-一般, D-较差
     */
    private String rating;

    /**
     * 状态：ACTIVE-正常, INACTIVE-停用
     */
    private String status;

    /**
     * 备注
     */
    private String remarks;
}

File: frontend/apps/web-antd/src/api/admin/supplier.ts
import type { PageResult } from '../matter/types';

/**
 * 供应商管理 API
 */
import { requestClient } from '#/api/request';

// ========== 类型定义 ==========
export interface SupplierDTO {
  id: number;
  supplierNo?: string;
  name: string;
  supplierType?: string;
  supplierTypeName?: string;
  contactPerson?: string;
  contactPhone?: string;
  contactEmail?: string;
  address?: string;
  creditCode?: string;
  bankName?: string;
  bankAccount?: string;
  supplyScope?: string;
  rating?: string;
  ratingName?: string;
  status?: string;
  statusName?: string;
  remarks?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface SupplierQuery {
  pageNum?: number;
  pageSize?: number;
  keyword?: string;
  supplierType?: string;
  status?: string;
  rating?: string;
}

export interface CreateSupplierCommand {
  name: string;
  supplierType?: string;
  contactPerson?: string;
  contactPhone?: string;
  contactEmail?: string;
  address?: string;
  creditCode?: string;
  bankName?: string;
  bankAccount?: string;
  supplyScope?: string;
  rating?: string;
  remarks?: string;
}

// ========== API 函数 ==========

/** 分页查询供应商列表 */
export function getSupplierList(params: SupplierQuery) {
  return requestClient.get<PageResult<SupplierDTO>>('/admin/suppliers', {
    params,
  });
}

/** 获取供应商详情 */
export function getSupplierDetail(id: number) {
  return requestClient.get<SupplierDTO>(`/admin/suppliers/${id}`);
}

/** 创建供应商 */
export function createSupplier(data: CreateSupplierCommand) {
  return requestClient.post<SupplierDTO>('/admin/suppliers', data);
}

/** 更新供应商 */
export function updateSupplier(id: number, data: CreateSupplierCommand) {
  return requestClient.put<SupplierDTO>(`/admin/suppliers/${id}`, data);
}

/** 删除供应商 */
export function deleteSupplier(id: number) {
  return requestClient.delete(`/admin/suppliers/${id}`);
}

/** 启用/停用供应商 */
export function changeSupplierStatus(id: number, status: string) {
  return requestClient.put(`/admin/suppliers/${id}/status`, null, {
    params: { status },
  });
}

/** 获取供应商统计 */
export function getSupplierStatistics() {
  return requestClient.get<Record<string, any>>('/admin/suppliers/statistics');
}
